### ISSUE TITLE
Notebook Document support is not implemented correctly

### ISSUE THREAD

**Issue by moucha19** [OmniSharp/csharp-language-server-protocol#1066]
Hello,

I was trying to connect from the LanguageClient to the **pylsp** server and when I did this with the current version of pylsp that supports LSP v. 3.17, the `Initialize` call ends in following Exception related to JSON Deserialization:

`Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'OmniSharp.Extensions.LanguageServer.Protocol.Models.NotebookSelector' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly`

 I did some digging and the culprit seems to be this part of the pylsp response that specifies server capabilities:

`'notebookDocumentSync': {'notebookSelector': [{'cells': [{'language': 'python'}]}]}`

As you can see, the top-level object of notebookSelector is array, which is (if I understand the LSP documentation correctly) a valid syntax, but the LanguageClient does not seem to support it.

**Comment by manandre** [comment]
The issue is not at `notebookSelector` level but at cell level. One `cell` only supports one `language`.
See the [notebookDocumentSyncOptions](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#notebookDocumentSyncOptions) specification.
```
cells: { language: string }[];
```
The payload here should be:
```
'notebookDocumentSync': {'notebookSelector': [{'cells': {'language': 'python'}}]}
```

**Comment by manandre** [comment]
The implementation is not aligned on the LSP 3.17 specification.
https://github.com/OmniSharp/csharp-language-server-protocol/blob/86dc4ac3db664b3973eda6525f681e23c62bc85a/src/Protocol/Features/Document/NotebookDocumentSyncFeature.cs#L757-L765
`NotebookSelector` should be handled as an array/container.

**Comment by manandre** [comment]
For information, `pylsp` have recently changed their implementation to comply to the specification.
See https://github.com/python-lsp/python-lsp-server/issues/453

**Comment by manandre** [comment]
LSP documentation fix: https://github.com/microsoft/language-server-protocol/pull/1831
Implementation fix: #1077
